neuralop/models/transformer_no.py

```diff
@@ -0,0 +1,170 @@
+        mlp_dropout : float , optional
+            droupout parameter of MLP layer, by default 0
+        mlp_expansion : float, optional
+            expansion parameter of MLP layer, by default 2.0
+        non_linearity : nn.Module, optional
+            Non-Linearity module to use, by default F.gelu
+        norm: string, optional
+            Normalization module to use, by default layernorm
+
+        """
+    def __init__(self,
+                 n_dim,
+                 in_channels=1,
+                 out_channels=1,
+                 encoder_hidden_channels=128,
+                 decoder_hidden_channels=128,
+                 encoder_num_heads=1,
+                 decoder_num_heads=8,
+                 encoder_head_n_channels=None,
+                 decoder_head_n_channels=None,
+                 encoder_n_layers=3,
+                 query_basis='siren',
+                 query_siren_layers=4,          # number of layers in SirenNet
+                 query_fourier_scale=2.0,       # scale of the Gaussian Fourier Feature Transform
+                 pos_emb='rotary',              # ['rotary', 'none']
+                 use_mlp=True,
+                 mlp_dropout=0,
+                 mlp_expansion=2.0,
+                 non_linearity=F.gelu,
+                 norm='layer_norm',      # ['layer_norm', 'instance_norm', ''group_norm', 'none']
+                ):
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.n_dim = n_dim
+        self.encoder_num_heads = encoder_num_heads
+        self.decoder_num_heads = decoder_num_heads
+        self.encoder_hidden_channels = encoder_hidden_channels
+        self.decoder_hidden_channels = decoder_hidden_channels
+        self.encoder_head_n_channels = encoder_head_n_channels if encoder_head_n_channels is not None else encoder_hidden_channels
+        self.decoder_head_n_channels = decoder_head_n_channels if decoder_head_n_channels is not None else decoder_hidden_channels
+        self.encoder_n_layers = encoder_n_layers
+        self.query_basis = query_basis
+        self.query_siren_layers = query_siren_layers
+        self.query_fourier_scale = query_fourier_scale
+        self.pos_emb = pos_emb
+
+        self.use_mlp = use_mlp
+        self.mlp_dropout = mlp_dropout
+        self.mlp_expansion = mlp_expansion
+        self.non_linearity = non_linearity
+        self.norm = norm
+
+        if self.pos_emb not in ['rotary', 'none']:
+            raise ValueError(f'pos_emb must be one of ["rotary", "none"], got {self.pos_emb}')
+
+        if self.pos_emb == 'rotary':
+            self.enc_pos_emb_module = RotaryEmbedding(self.encoder_head_n_channels // self.n_dim)
+            self.dec_pos_emb_module = RotaryEmbedding(self.decoder_head_n_channels // self.n_dim)
+        else:
+            self.enc_pos_emb_module = None
+            self.dec_pos_emb_module = None
+
+        self.encoder = nn.ModuleList(
```

Comment:
nit: `FNOBlocks` takes `n_layers` as a parameter instead of exposing a moduleList within the `FNO` itself. It might be nice to have a similar convention here

Reply:
Got it;)